File: ABC/351-400/381/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<n;++i)

int main(){
    int n;
    cin >> n;
    string s;
    cin >> s;
    vector<int> a;
    rep(i,n) if(s[i] == '/') a.push_back(i);
    int ans = 0;
    vector<bool> seen(n);
    for(int x : a){
        if(seen[x]) continue;
        int l = x-1;
        int r = x+1;
        seen[x] = true;
        while(l>=0 && s[l] == '1'){
            seen[l] = true;
            l--;
        } 
        while(r<n&&s[r] =='2'){
            seen[r] = true;
            r++;
        }
        l++;
        r--;
        int now = min(x-l,r-x);
        ans = max(ans,now*2+1);
    }
    cout << ans << endl;
    return 0;
}

File: ABC/351-400/381/c.py
n = int(input())
s = input()

ans = 0

for i in range(n):
    if s[i] == '/':
        l = i-1
        r = i+1
        while l >= 0 and s[l] == '1':
            l -=1
        while r < n and s[r] == '2':
            r += 1
        l += 1
        r -= 1
        now = min(max(r-i,0),max(i-l,0))
        ans = max(now*2+1,ans)
        
print(ans)